dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/logging/source/AccessLogSource.java

```diff
@@ -56,33 +59,82 @@
     private final String location = "memory";
     private static String USER_AGENT_HEADER = "User-Agent";
     public static final int MAX_USER_AGENT_LENGTH = 2048;
-    private SetterFormatter currentSF = new SetterFormatter();
-    public String jsonAccessLogFields = "";
-    public static String jsonAccessLogFieldsLogstash = "";
+    Map<Configuration, SetterFormatter> setterFormatterMap = new ConcurrentHashMap<Configuration, SetterFormatter>();
+    private SetterFormatter currentSF = new SetterFormatter("", "", "");
+    public String jsonAccessLogFieldsConfig = "";
+    public static String jsonAccessLogFieldsLogstashConfig = "";
     public Map<String, Object> configuration;
 
+    // A representation of the current configuration; to be used in the setterFormatterMap
+    private class Configuration {
+        String logFormat;
+        String loggingConfig;
+        String logstashConfig;
+
+        private Configuration(String logFormat, String loggingConfig, String logstashConfig) {
+            this.logFormat = logFormat;
+            this.loggingConfig = loggingConfig;
+            this.logstashConfig = logstashConfig;
+        }
+
+        //@formatter:off
+        String getLogFormat()      { return this.logFormat; }
+        String getLoggingConfig()  { return this.loggingConfig; }
+        String getLogstashConfig() { return this.logstashConfig; }
+        //@formatter:on
+
+        // We need to put this object into a HashMap, so we're overriding hashCode
+        @Override
+        public int hashCode() {
+            int hash;
+            hash = logFormat.hashCode() * loggingConfig.hashCode() * logstashConfig.hashCode();
+            return hash;
+        }
+
+        @Override
+        public boolean equals(Object obj) {
+            if (this == obj)
+                return true;
+            if (obj == null)
+                return false;
+            if (getClass() != obj.getClass())
+                return false;
+            Configuration config = (Configuration) obj;
+            if (!config.getLogFormat().equals(this.logFormat) || !config.getLoggingConfig().equals(this.loggingConfig) || !config.getLogstashConfig().equals(this.logstashConfig))
+                return false;
+            return true;
+        }
+    }
+
     private static class SetterFormatter {
+        // The HTTP access logging logFormat value, e.g. "%a %b"
         String logFormat;
+        // The jsonAccessLogFields configuration value for JSON logging, default or logFormat
         String loggingConfig;
+        // The jsonAccessLogFields configuration value for Logstash Collector, default or logFormat
         String logstashConfig;
```

Comment:
Might make sense to use the Configuration class here instead of having the same 3 fields in SetterFormatter